mlflow/tracking/fluent.py

```diff
@@ -1233,36 +1240,71 @@ def log_image(image: Union["numpy.ndarray", "PIL.Image.Image"], artifact_file: s
             - H x W x 4 (an RGBA channel order is assumed)
 
     Args:
-        image: Image to log.
-        artifact_file: The run-relative artifact file path in posixpath format to which
-            the image is saved (e.g. "dir/image.png").
+        key: Image name. This string may only contain alphanumerics, underscores
+            (_), dashes (-), periods (.), spaces ( ), and slashes (/).
+            All backend stores will support keys up to length 250, but some may
+            support larger keys.
+        image: The image object to be logged.
+        step: Integer training step (iteration) at which the image was saved.
+            Defaults to 0.
+        timestamp: Time when this image was saved. Defaults to the current system time.
+        artifact_file: Legacy. Specifies the path, in POSIX format, where the image
+            will be stored as an artifact relative to the run's root directory (for
+            example, "dir/image.png"). This parameter is kept for backward compatibility
+            and should not be used together with `key`, `step`, or `timestamp`.
+        *args: Positional arguments placeholder. Please refer to the documentation above.
+        **kwargs: Keyword arguments placeholder. Please refer to the documentation above.
 
     .. code-block:: python
-        :test:
-        :caption: Numpy Example
+        :caption: Time-stepped image logging numpy example
 
         import mlflow
         import numpy as np
 
         image = np.random.randint(0, 256, size=(100, 100, 3), dtype=np.uint8)
 
-        with mlflow.start_run():
-            mlflow.log_image(image, "image.png")
+        run = mlflow.create_run(experiment_id="0")
```

Comment:
this code will leave a hanging mlflow run, let's use:
`with mlflow.start_run() as run` instead.